### PR TITLE
search players in DB

### PR DESCRIPTION
**Pull request by SolatK**
## About The Pull Request

изображение
изображение
изображение

## Why It's Good For The Game

Why not?

## Changelog
:cl:
add: The server now have whitelist  for VPN checking.
add: Admins now can search players in the server DB.
add: server now check if a player have same ip or cid with somebody else from BD and remember it FOREVER.
fix: delayed ban kick player
/:cl:

### CODE REVIEW

**Comment by SpadesSlick** [comment]
Awaiting for refresh fix.

**Comment by SolatK** [comment]
> 
> 
> Awaiting for refresh fix.

Fixed

**Comment by SolatK** [comment]
STILL WIP!

**Comment by SolatK** [comment]
Pls review

**Comment by TorinoFermic** [comment]
I don't see any sanitize functions for search terms.

File: code/modules/admin/DB_search/search.dm [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	var/empty = 1
+
+/datum/DB_search/verb/new_search_related(var/ckey as text)
+	set category = "Admin"
+	set name = "Search related accounts"
+	set desc = "Search players with same IP or CID"
+
+	var/list/ip_related_ckeys = list()
+	var/list/cid_related_ckeys = list()
+	var/DBQuery/search_query = dbcon.NewQuery("SELECT ip_related_ids, cid_related_ids FROM players WHERE ckey = '[ckey]'")
```
Comment (ACCount12) [inline code/modules/admin/DB_search/search.dm:16]: `ckey` is unsanitized input.

File: code/modules/admin/DB_search/search.dm [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		<th>IP</th>
+		<th>CID</th>
+		<th>last online</th>
+	</tr>"}
+
+		hsrc.panel.set_content(output)
+		hsrc.empty = 1
+	if(dbsearchckey_search || dbsearchip_search || dbsearchcid_search)
+		hsrc.empty = 0
+		var/DBQuery/search_query = dbcon.NewQuery("SELECT ckey, ip, cid, last_seen FROM players WHERE ckey = '[dbsearchckey_search]' OR ip = '[dbsearchip_search]' OR cid = '[dbsearchcid_search]'")
```
Comment (ACCount12) [inline code/modules/admin/DB_search/search.dm:120]: `dbsearchckey_search`, `dbsearchip_search` and `dbsearchcid_search` are unsanitized inputs.

File: code/modules/admin/verbs/antiraid.dm [hunk 3]
```diff
@@ -43,3 +43,25 @@ ADMIN_VERB_ADD(/client/proc/ip_reputation, R_ADMIN, FALSE)
...
+	set name = "Toggle VPN Checks"
+
+	if(!check_rights(R_ADMIN))
+		return
+
+	if (!dbcon || !dbcon.IsConnected())
+		to_chat(usr,"The database is not connected!")
+		return
+
+	var/DBQuery/query = dbcon.NewQuery("SELECT id FROM players WHERE ckey = '[ckey]'")
```
Comment (ACCount12) [inline code/modules/admin/verbs/antiraid.dm:59]: `ckey` is unsanitized input.

**Pull request merged**